tests/test_market.py

```diff
@@ -0,0 +1,423 @@
+        # simulation is over, now we inspect the output of the agents' wallets
+        number_of_init_agents = 0  # count number of init agents so we can skip over them
+        for all_agent_index, agent in simulator.agents.items():  # loop over all agents
+            if agent.name == "init_lp":
+                number_of_init_agents += 1
+            else:  # only for custom agents, loop across them and check their parameters
+                custom_agent_index = all_agent_index - number_of_init_agents  # identify which custom agent we are on
+                expected_wallet_dict = expected_wallet[custom_agent_index]
+                for account, expected_balance in expected_wallet_dict.items():  # for each custom parameter to check
+                    agent_balance_in_pts = getattr(agent.wallet, account)
+                    if isinstance(agent_balance_in_pts, dict):
+                        list_of_account_values = list(agent_balance_in_pts.values())
+                        agent_balance_in_pts = sum(x.balance for x in list_of_account_values)
+                    agent_balance_in_base = agent_balance_in_pts * simulator.market.spot_price
+
+                    ##### GENERAL LOGIC TESTING #####
+                    # value of Long position in base has to be less than the # of PTs
+                    # it could only get close if price were equal to 1, and even then there would be slippage
+                    assert agent_balance_in_base < agent.amount_to_trade, (
+                        f"agent #{custom_agent_index}'s {account} is worth more than the amount they traded",
+                    )
+
+                    ##### SPECIFIC MANUAL TESTING #####
+                    # this holds only if we know the exact value
+                    np.testing.assert_almost_equal(
+                        agent_balance_in_pts,
+                        expected_balance,  # passed in as pt's
+                        err_msg=(
+                            f"agent #{custom_agent_index}'s {account} does not equal {expected_balance}"
+                            f", instead we have (base,pt)=({agent_balance_in_base},{agent_balance_in_pts})"
+                        ),
+                    )
+
+    def assert_equal_and_log(self, thing: str, expected: Any, actual: Any):
+        """Compare actual thing to expected thing"""
+        logging.debug("=== %s ===\nexpected %s: %s\nactual %s: %s", thing.upper(), thing, expected, thing, actual)
+        self.assertEqual(expected, actual, f"{thing} do not match")
+
+    def compare_deltas(self, actual_deltas: Deltas, expected_deltas: Deltas):
+        """Compare actual deltas to expected deltas"""
+        self.assert_equal_and_log("market deltas", expected_deltas.market_deltas, actual_deltas.market_deltas)
+        self.assert_equal_and_log("agent deltas", expected_deltas.agent_deltas, actual_deltas.agent_deltas)
+
+    def compare_wallets(self, actual_wallet: Wallet, expected_wallet: Wallet):
```

Comment:
This is defined and then never used (just referenced in a commented line). I think it would be better to just comment out line 113: `self.assert_equal_and_log("wallet", expected_wallet, actual_wallet)` below and delete this function.